app/server/lib/Assistance.ts

```diff
@@ -117,23 +117,54 @@ class RetryableError extends Error {
 }
 
 /**
- * A flavor of assistant for use with the OpenAI API.
+ * A flavor of assistant for use with the OpenAI chat completion endpoint
+ * and tools with a compatible endpoint (e.g. llama-cpp-python).
  * Tested primarily with gpt-3.5-turbo.
+ *
+ * Uses the ASSISTANT_CHAT_COMPLETION_ENDPOINT endpoint if set, else
+ * an OpenAI endpoint. Passes ASSISTANT_API_KEY or OPENAI_API_KEY in
+ * a header if set. An api key is required for the default OpenAI
+ * endpoint.
+ *
+ * If a model string is set in ASSISTANT_MODEL, this will be passed
+ * along. For the default OpenAI endpoint, a gpt-3.5-turbo variant
+ * will be set by default.
+ *
+ * If a request fails because of context length limitation, and the
+ * default OpenAI endpoint is in use, the request will be retried
+ * with ASSISTANT_LONGER_CONTEXT_MODEL (another gpt-3.5
+ * variant by default). Set this variable to "" if this behavior is
+ * not desired for the default OpenAI endpoint. If a custom endpoint was
+ * provided, this behavior will only happen if
+ * ASSISTANT_LONGER_CONTEXT_MODEL is explicitly set.
+ *
+ * An optional ASSISTANT_MAX_TOKENS can be specified.
  */
 export class OpenAIAssistant implements Assistant {
   public static DEFAULT_MODEL = "gpt-3.5-turbo-0613";
-  public static LONGER_CONTEXT_MODEL = "gpt-3.5-turbo-16k-0613";
+  public static DEFAULT_LONGER_CONTEXT_MODEL = "gpt-3.5-turbo-16k-0613";
+  public static MAX_TOKENS = process.env.ASSISTANT_MAX_TOKENS ?
```

Comment:
optional: feels like this should be an instance field like the others for consistency